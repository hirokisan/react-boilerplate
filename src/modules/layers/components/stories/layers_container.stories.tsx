import React from 'react';

import { LayersContainer } from '../layers_container';

export default {
  title: 'Layers'
};

const Wrapper = () => <LayersContainer>werwer 12012</LayersContainer>;

export const LayerContainer = () => <Wrapper />;

LayerContainer.story = {
  name: 'default'
};
