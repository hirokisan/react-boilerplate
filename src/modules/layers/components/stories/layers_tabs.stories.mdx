import { action } from '@storybook/addon-actions';
import { Story, Meta, Props, Source } from '@storybook/addon-docs/blocks';
import { text } from '@storybook/addon-knobs';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { LayersWrapper } from '../layers_container';

<Meta 
  title='Components|Layers/Tabs' 
  component={LayersTabs} 
  decorators={[(storyFn) => <LayersWrapper>{storyFn()}</LayersWrapper>]} 
/>

# Layer Tabs

Layer tabs switch between different layer groups

### Example

<Story name="first selected">
  <LayersTabs
    currentTab={number('ActiveTab', 0)}
    setTab={action('setTab')}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
</Story>

### Props

<Props of={LayersTabs} />

## Stories

### Second Selected

<Story name="second selected">
  <LayersTabs
    currentTab={1}
    setTab={action('setTab')}
    tabs={['LAYERS', 'INDICATORS']}
  />
</Story>