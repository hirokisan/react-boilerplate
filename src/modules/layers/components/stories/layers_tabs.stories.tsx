import React from 'react';
import { action } from '@storybook/addon-actions';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { LayersWrapper } from '../layers_container';
import { Action } from 'config/common';

export default {
  title: 'Components|Layers/Tabs',
  parameters: {
    component: LayersTabs,
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <LayersWrapper>{storyFn()}</LayersWrapper>]
  }
};

const storyProps = {
  setTab: action('setTab') as Action<number>
};

export const DefaultView = ({ setTab } = storyProps) => (
  <LayersTabs
    currentTab={number('ActiveTab', 0)}
    setTab={setTab}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
);

DefaultView.story = {
  name: 'first selected'
};

export const SelectedView = ({ setTab } = storyProps) => (
  <LayersTabs currentTab={1} setTab={setTab} tabs={['LAYERS', 'INDICATORS']} />
);

SelectedView.story = {
  name: 'second selected'
};
