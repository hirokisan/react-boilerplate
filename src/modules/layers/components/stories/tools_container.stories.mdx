import { action } from '@storybook/addon-actions';
import { Story, Meta, Props, Source } from '@storybook/addon-docs/blocks';
import { text } from '@storybook/addon-knobs';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { ToolsContainer } from '../tools_container';
import { LayersWrapper } from '../layers_container';

<Meta 
  title='Components|Layers/Tools' 
  component={ToolsContainer} 
  decorators={[(storyFn) => <LayersWrapper>{storyFn()}</LayersWrapper>]} 
/>

# Tools

Set of tools to operate pane content (e.g. layers)

### Example

<Story name="first selected">
  <ToolsContainer setTool={action('setTool')} />
</Story>

### Props

<Props of={ToolsContainer} />


