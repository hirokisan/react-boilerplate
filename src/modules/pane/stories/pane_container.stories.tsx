import React from 'react';

import { ToolsContainer } from '../pane_tools';
import { PaneContainer } from '../pane_container';

import marked from 'marked';
import { PaneTabs } from '../pane_tabs';
import { action } from '@storybook/addon-actions';

const docs = `
The layer container is one of the core system components, allowing user to display and manipulate layers. 
`;

export default {
  title: 'Components|Pane/Container',
  parameters: {
    component: ToolsContainer,
    info: marked(docs)
  }
};

export const DefaultView = () => <PaneContainer>This is some content</PaneContainer>;

DefaultView.story = {
  name: 'default'
};

const Story = () => {
  const [tab, setTab] = React.useState(0);
  return (
    <PaneContainer>
      <PaneTabs setTab={setTab} currentTab={tab} tabs={['LAYERS', 'ACTIONS']} />
      <ToolsContainer setTool={action('setTool')} />
      <div style={{ padding: '6px', color: 'white' }}>
        {tab == 0 && <div>Tab LAYERS</div>}
        {tab == 1 && <div>Tab ACTIONS</div>}
      </div>
    </PaneContainer>
  );
};

// weird bug not allowing to render hooks directly in the story
// we just use a sub component
export const CompoundView = () => {
  return <Story />;
};

CompoundView.story = {
  name: 'compound view'
};
