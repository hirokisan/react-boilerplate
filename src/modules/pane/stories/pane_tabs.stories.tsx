import React from 'react';
import { action } from '@storybook/addon-actions';
import { PaneTabs } from '../pane_tabs';
import { array, number } from '@storybook/addon-knobs';
import { PaneContainer } from '../pane_container';
import { Action } from 'config/common';

export default {
  title: 'Components|Pane/Tabs',
  parameters: {
    component: PaneTabs,
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <PaneContainer>{storyFn()}</PaneContainer>]
  }
};

const storyProps = {
  setTab: action('setTab') as Action<number>
};

export const DefaultView = ({ setTab } = storyProps) => (
  <PaneTabs
    currentTab={number('ActiveTab', 0)}
    setTab={setTab}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
);

DefaultView.story = {
  name: 'first selected'
};

export const SelectedView = ({ setTab } = storyProps) => (
  <PaneTabs currentTab={1} setTab={setTab} tabs={['LAYERS', 'INDICATORS']} />
);

SelectedView.story = {
  name: 'second selected'
};
