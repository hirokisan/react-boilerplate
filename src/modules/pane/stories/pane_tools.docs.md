# Title

This is some text in an advanced Markdown specification
