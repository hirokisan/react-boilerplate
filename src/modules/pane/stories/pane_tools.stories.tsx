import React from 'react';

import { action } from '@storybook/addon-actions';
import { ToolsContainer } from '../pane_tools';
import { PaneContainer } from '../pane_container';

import docs from './pane_tools.docs.md';
import marked from 'marked';
import { Action } from 'config/common';

export default {
  title: 'Components|Pane/Tools',
  parameters: {
    component: ToolsContainer,
    info: marked(docs),
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <PaneContainer>{storyFn()}</PaneContainer>]
  }
};

export const DefaultView = (setToolHandler?: Action<string>) => (
  <ToolsContainer setTool={setToolHandler || action('setTool')} />
);

DefaultView.story = {
  name: 'default'
};
