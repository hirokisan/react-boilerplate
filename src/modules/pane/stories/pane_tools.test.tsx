import { render, fireEvent } from '@testing-library/react';

import { DefaultView } from './pane_tools.stories';

describe('Tools View', () => {
  const setTool = jest.fn();

  it('renders default', () => {
    const { container } = render(DefaultView(setTool));
    expect(container).toMatchSnapshot();
  });

  it('selects another item and calls a callback function', () => {
    const { getByTitle } = render(DefaultView(setTool));
    fireEvent.click(getByTitle('edit'));
    expect(getByTitle('edit')).toHaveClass('active');
    expect(setTool).toHaveBeenCalledWith('edit');
  });
});
