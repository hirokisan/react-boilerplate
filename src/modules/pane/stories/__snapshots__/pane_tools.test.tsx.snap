// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools View renders default 1`] = `
.emotion-6 {
  width: 100%;
  height: 40px;
  min-height: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-1 {
  width: 40px;
  height: 30px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #000000;
  border-radius: 2px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 10px;
}

.emotion-1:hover,
.emotion-1.active {
  background-color: #000000;
}

.emotion-1 .icon {
  width: 18px;
}

.emotion-0 {
  width: 18px;
}

<div>
  <ul
    class="emotion-6"
  >
    <li
      class="active emotion-1"
      data-tool="select"
      title="select"
    >
      <img
        alt="select"
        class="emotion-0"
        src="https://deliverytestdata.blob.core.windows.net/deliverytest/cursor.svg"
      />
    </li>
    <li
      class=" emotion-1"
      data-tool="edit"
      title="edit"
    >
      <img
        alt="edit"
        class="emotion-0"
        src="https://deliverytestdata.blob.core.windows.net/deliverytest/pluma.svg"
      />
    </li>
    <li
      class=" emotion-1"
      data-tool="measure"
      title="measure"
    >
      <img
        alt="measure"
        class="emotion-0"
        src="https://deliverytestdata.blob.core.windows.net/deliverytest/measure.svg"
      />
    </li>
  </ul>
</div>
`;
